flink-formats/flink-csv/src/test/java/org/apache/flink/formats/csv/CsvRowDeSerializationSchemaTest.java

```diff
@@ -0,0 +1,223 @@
+			new BigInteger("123400000000000000000000000000"));
+		testNullableField(Types.SQL_DATE, "2018-10-12", Date.valueOf("2018-10-12"));
+		testNullableField(Types.SQL_TIME, "12:12:12", Time.valueOf("12:12:12"));
+		testNullableField(
+			Types.SQL_TIMESTAMP,
+			"\"2018-10-12 12:12:12.0\"",
+			Timestamp.valueOf("2018-10-12 12:12:12"));
+		testNullableField(
+			Types.ROW(Types.STRING, Types.INT, Types.BOOLEAN),
+			"Hello;42;false",
+			Row.of("Hello", 42, false));
+		testNullableField(
+			Types.OBJECT_ARRAY(Types.STRING),
+			"a;b;c",
+			new String[] {"a", "b", "c"});
+		testNullableField(
+			Types.OBJECT_ARRAY(Types.BYTE),
+			"12;4;null",
+			new Byte[] {12, 4, null});
+		testNullableField(
+			(TypeInformation<byte[]>) Types.PRIMITIVE_ARRAY(Types.BYTE),
+			"awML",
+			new byte[] {107, 3, 11});
+	}
+
+	@Test
+	public void testCustomizedProperties() throws IOException {
+
+		final Consumer<CsvRowSerializationSchema> serConfig = (serSchema) -> {
+			serSchema.setEscapeCharacter('*');
+			serSchema.setQuoteCharacter('\'');
+			serSchema.setArrayElementDelimiter(":");
+			serSchema.setFieldDelimiter(';');
+		};
+
+		final Consumer<CsvRowDeserializationSchema> deserConfig = (deserSchema) -> {
+			deserSchema.setEscapeCharacter('*');
+			deserSchema.setQuoteCharacter('\'');
+			deserSchema.setArrayElementDelimiter(":");
+			deserSchema.setFieldDelimiter(';');
+		};
+
+		testField(Types.STRING, "123*'4**", "123'4*", deserConfig, ";");
+		testField(Types.STRING, "'123''4**'", "123'4*", serConfig, deserConfig, ";");
+		testField(Types.STRING, "'a;b*'c'", "a;b'c", deserConfig, ";");
+		testField(Types.STRING, "'a;b''c'", "a;b'c", serConfig, deserConfig, ";");
+		testField(Types.INT, "       12          ", 12, deserConfig, ";");
+		testField(Types.INT, "12", 12, serConfig, deserConfig, ";");
+		testField(Types.ROW(Types.STRING, Types.STRING), "1:hello", Row.of("1", "hello"), deserConfig, ";");
+		testField(Types.ROW(Types.STRING, Types.STRING), "'1:hello'", Row.of("1", "hello"), serConfig, deserConfig, ";");
+		testField(Types.ROW(Types.STRING, Types.STRING), "'1:hello world'", Row.of("1", "hello world"), serConfig, deserConfig, ";");
+		testField(Types.STRING, "null", "null", serConfig, deserConfig, ";"); // string because null literal has not been set
+	}
+
+	@Test
+	public void testDeserializationProperties() throws IOException {
+		final TypeInformation<Row> rowInfo = Types.ROW(Types.STRING, Types.INT, Types.STRING);
+		final CsvRowDeserializationSchema deserializationSchema = new CsvRowDeserializationSchema(rowInfo);
+
+		try {
+			deserializationSchema.deserialize("Test,null,Test".getBytes()); // null not supported
+			fail("Missing field should cause failure.");
+		} catch (IOException e) {
+			// valid exception
+		}
+
+		deserializationSchema.setIgnoreParseErrors(true);
+
+		// unsupported null for integer
+		assertEquals(Row.of("Test", null, "Test"), deserializationSchema.deserialize("Test,null,Test".getBytes()));
```

Comment:
nit: Could we have those as separate tests? Personally prefer this approach as it immediately shows what failed (by the test name).